validation-tool-backend/src/users/users.service.ts

```diff
@@ -74,7 +74,18 @@ export class UsersService {
         AND password = HASHBYTES('SHA2_256', @Password)`);
     if (result.recordset && result.recordset.length > 0) {
       user = result.recordset[0] as User;
```

Comment:
Array destructuring can be used here. 

Also, assigning `result.recordset` to a variable would be better than accessing the properties every time. With this, you could also set the type User to this variable instead of doing so to the individual `user` one. 

```
const userArray: User[] = result.recordset;

if (userArray && userArray.length) {
    const [user] = userArray;
    ...
}
```